app/controllers/posts_controller.rb

```diff
@@ -9,11 +9,12 @@ def index
   # The function assigns variables for a post, comment, like, and user in order
   # to display them in a view.
   def show
-    @post = Post.find(params[:id])
+    @user = current_user
+    @post = Post.find_by(id: params[:id])
```

Comment:
- [ ] add this `load_and_authorize_resource` at the top of your class to authorize posts